Nodejs/Product/Nodejs/Project/NodeModulesNode.cs

```diff
@@ -362,9 +377,9 @@ private class NpmPathProvider : INpmPathProvider {
             }
         }
 
-#endregion
+        #endregion
 
-#region Updating module hierarchy
+        #region Updating module hierarchy
 
         internal void ReloadHierarchySafe() {
```

Comment:
Calling the typings acquisition code at ReloadHierarchy will cause serious performance issues because it will operates on the UI thread and gets called far more often than necessary for our purposes. Instead, move this logic to `NodejsProjectNode.OnIdleNodeModules`. Additionally, from `NodeModules`, you can add a method to GetTopLevelDependencies that you populate via `AddTopLevelModule`